src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1645,6 +1645,35 @@ public bool Equals(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Determines if the contents of this builder are equal to the contents of ReadOnlySpan<char>.
+        /// </summary>
+        /// <param name="value">The ReadOnlySpan{char}.</param>
+        public bool Equals(ReadOnlySpan<char> value)
+        {
+            if (value.Length != Length)
+                return false;
+
+            StringBuilder sbChunk = this;
+            int offset = 0;
+
+            do
+            {
+                int chunk_length = sbChunk.m_ChunkLength;
+                offset += chunk_length;
+
+                ReadOnlySpan<char> chunk = new ReadOnlySpan<char>(sbChunk.m_ChunkChars, 0, chunk_length);
+
+                if (!StringSpanHelpers.Equals(chunk, value.Slice(value.Length - offset, chunk_length)))
```

Comment:
this is an extension method, you can do 
`chunk.Equals(value.Slice(value.Length - offset, chunk_length))))`